server/src/tests/Feature/hookSystemTest.php

```diff
@@ -0,0 +1,90 @@
+{
+    use WithFaker;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    /**
+     * Actions listen to an Event, so when an event is triggered, an Action that is listening
+     * to that event will also trigger. The third party developer will register Actions for an Event. These Actions will Listen
+     * for the Event to fire and then execute.
+     *
+     * @return void
+     */
+    public function testEventService()
+    {
+        // Set actions for event at run time, for testing purposes
+        config(['events.' . BeforeCheckout::class => [StripeBeforeCheckout::class, FooBeforeCheckout::class, BarBeforeCheckout::class]]);
+
+        // Trigger the event through EventService
+        $eventService = EventService::triggerEvent(BeforeCheckout::class);
+
+        // All the actions that we're called
+        $actions = $eventService->getMeta();
+
+        // Prepare array for essertContains (flatten array)
+        $actions = array_column($actions, 'action');
+
+        // Check that action names were added to the meta
+        $this->assertContains(StripeBeforeCheckout::class, $actions, "actionsArray doesn't contain StripeBeforeCheckout");
+        $this->assertContains(FooBeforeCheckout::class, $actions, "actionsArray doesn't contain FooBeforeCheckout");
+        $this->assertContains(BarBeforeCheckout::class, $actions, "actionsArray doesn't contain BarBeforeCheckout");
+    }
+
+    public function testEventServiceWithData()
+    {
+        // Some arbitrary data used as an example of what a third party may need, and then change
+        $checkoutData = [
+            "name" => "Luke",
+            "email" => "cart@example.com",
+            "telephone" => "082456748",
+            "mobile" => "08357684758",
+            "billing_address" => [
+                "line_1" => "1 London Way",
+                "line_2" => "",
+                "state" => "London",
+                "postcode" => "SW1A 1AA",
+                "country" => "UK"
+            ],
+            "shipping_address" => [
+                "line_1" => "1 London Way",
+                "line_2" => "",
+                "state" => "London",
+                "postcode" => "SW1A 1AA",
+                "country" => "UK"
+            ],
+            "use_shipping_for_billing" => true,
+            "payment_method" => "",
+            "agree_to_terms" => true,
+            "notes" => ""
+        ];
+
+        // Set actions for event at run time, for testing purposes
+        config(['events.' . BeforeCheckout::class => [StripeBeforeCheckout::class, FooBeforeCheckout::class, BarBeforeCheckout::class]]);
+
+        // Trigger the event through EventService
+        $eventService = EventService::triggerEvent(BeforeCheckout::class, $checkoutData);
+
+        $historyOfModifiedData = $eventService->getLastModifiedData();
+
+        $historyOfAllModifiedData = $eventService->getAllModifiedData();
+
+        $this->assertNotEmpty($historyOfModifiedData);
```

Comment:
@AlexeiaW I think we just need to focus testing on making sure data is coming out correctly, for example, if a few actions were used for input, make sure they are all reflected in the output. Otherwise there may be some data being overwritten in code we aren't aware of for e.g.